src/paho/mqtt/client.py

```diff
@@ -1315,11 +1508,12 @@ def username_pw_set(self, username, password=None):
 
         # [MQTT-3.1.3-11] User name must be UTF-8 encoded string
         self._username = None if username is None else username.encode('utf-8')
-        self._password = password
-        if isinstance(self._password, str):
-            self._password = self._password.encode('utf-8')
+        if isinstance(password, str):
+            self._password = password.encode('utf-8')
+        else:
+            self._password = password
```

Comment:
This idiom should probably be refactored into a free helper function. (I've seen it called `force_bytes()` in e.g. Django, with `force_str()` being the counterpart.)